Dagger injection

JamesSanga opened the pull request:
Moviestore app that have been integrated with navigation components

JamesSanga committed:
Load next page added

JamesSanga committed:
Coding for the time needed to load next page movies

JamesSanga committed:
Method that will be triggered once next page has been loaded already so the loading footer should stop showing

JamesSanga committed:
importing host presenter

JamesSanga committed:
Arrangement

JamesSanga committed:
Implementing pull to refresh in first fragment

JamesSanga committed:
Updating move id in presenter

JamesSanga committed:
Pull to refresh

JamesSanga committed:
Pull to refresh movies in host fragment

JamesSanga committed:
Implementing what to do once swipe to refresh has been taken action

JamesSanga committed:
Checking current page if is equal to last page and allow to load more pages if current page is less than last page

JamesSanga committed:
Methods to update view if last page is equal or not to current page

JamesSanga committed:
Implementing dimens for large phones

JamesSanga committed:
Removing navigation to first fragment itself

JamesSanga committed:
Adding reload listener for related movies once are clicked so that can be updated with their data

JamesSanga committed:
dimens for tablet

JamesSanga committed:
Dimens for tablet a7

JamesSanga committed:
Dimens for typical phones(smallest phones)

JamesSanga committed:
Changing dimens height of different devices

JamesSanga committed:
Deleted

JamesSanga committed:
Deleted

JamesSanga committed:
Arranged

JamesSanga committed:
scroll view added and layout changed with the whole user interface

JamesSanga committed:
Implementation of additional user interface in first fragment

JamesSanga committed:
Removed api in host fragment due to its implementation has taken place in host presenter

JamesSanga committed:
One argument added that is date required in date field in move details

JamesSanga committed:
Putting date bundle in adapter to details fragment

JamesSanga committed:
Strings for similar movies and favorite

JamesSanga committed:
Trailer back ground img

JamesSanga committed:
Removed nested scroll view layout to allow user interface to take place

JamesSanga committed:
Setting app bar height

JamesSanga committed:
Adding to toolbar in bottom bar so that can be used as action bar

JamesSanga committed:
Changing height of the trailer to 250 up to 350 for large devices

JamesSanga committed:
nothing

JamesSanga committed:
New fragment to view favorite movies separate from movie details fragment

JamesSanga committed:
Implementation of movies trailer

JamesSanga committed:
User interface for favorite movies

JamesSanga committed:
Recycler view for trailer movies and favorite icon to add movies in favorite

JamesSanga committed:
Adding id to layout tha has been used to get view for navigation controller

JamesSanga committed:
Implementing route from host fragment to favorite fragment

JamesSanga committed:
Adding menu for favorite navigation

JamesSanga committed:
image for trailer movies

JamesSanga committed:
Getting date for movie to display in movie in movie details

JamesSanga committed:
Auto generated

JamesSanga committed:
Adding action route from host fragment to favorite movies

JamesSanga committed:
Service api for trailer movies

JamesSanga committed:
Strings added

JamesSanga committed:
image that appear over trailer

JamesSanga committed:
Trailer objects are converted into that string so that can be transferred and stored

JamesSanga committed:
Card view for movie trailer

JamesSanga committed:
Trailer adapter added

JamesSanga committed:
trailer response created

JamesSanga committed:
Removing commented lines of codes

JamesSanga committed:
Codding Grid layout manager

JamesSanga committed:
removed unused codes

JamesSanga committed:
Setting navigation host default layout constraint bottom tob be parent

JamesSanga committed:
Make some of global variables to be local variables so that to reduce warnings

JamesSanga committed:
Deleted classes due to be not used

JamesSanga committed:
Setting global variables to be local variables, removing methods that used to call methods from presenter instead presenter is called direct and work arrangement

JamesSanga committed:
Implementing movie trailers to mvp

JamesSanga committed:
trailer movies results interface method that trigger the changes in first fragment where trailer appears

JamesSanga committed:
Reducing congestion in onCreate method by adding other methods to work for other response like tool bar method

JamesSanga committed:
Removing double adding data in adapter when similar movies re-clicked several times

JamesSanga committed:
Bug solving by the way

JamesSanga committed:
Menu for favorite movies that allows a user to clear favorite moves

JamesSanga committed:
Setting up a clear view of display view (Favorite movies container)

JamesSanga committed:
Setting up the function of menu to display, adding swipe to refresh favorite movies and implementing method to delete all favorite movies from the SQlite database

JamesSanga committed:
Swipe to refresh layout in view and changed layout from constraint to relative layout

JamesSanga committed:
Bold movie details title

JamesSanga committed:
Disabling som of views in related movies container, adding margin, increasing card elevation, changing image scale type (center crop) and setting move tittle width to match parent

JamesSanga committed:
Library v4:28

JamesSanga committed:
Allowing auto width in any device

JamesSanga committed:
Displaying message to user when trying to clear empty list of movies

JamesSanga committed:
Adding method that loads more pages in pagination

JamesSanga committed:
Re implementing load more pages in presenter

JamesSanga committed:
Removing un used method

JamesSanga committed:
User interface for movie trailer

JamesSanga committed:
Permissions for budge notification and declaring trailer youtube player in manifest

JamesSanga committed:
Created customizable toolbar that is usable in youtube player activity

JamesSanga committed:
Updating some of library to current version

JamesSanga committed:
Integrating http network error with retrofit 2

JamesSanga committed:
Cleaning unnecessary line of codes

JamesSanga committed:
Adding construct in first presenter

JamesSanga committed:
Removing text set from exml and make height wrap content instead of fixed height

JamesSanga committed:
Setting up youtube player view

JamesSanga committed:
Now servies is used in presenter not in fragment

JamesSanga committed:
Adding construct for context defining error codes for network

JamesSanga committed:
Adding construct in first presenter

JamesSanga committed:
Drawable to cancel youtube player activity

JamesSanga committed:
Removing tag

JamesSanga committed:
Adding path in navigation graph

JamesSanga committed:
The page that check if the device is connected to internet or not

JamesSanga committed:
Defining message when the device is out of network

JamesSanga committed:
Solving the error of having null data due to un similar words used in the same work

JamesSanga committed:
Passing data to youtube player so that can be used to play trailer

JamesSanga committed:
Fragment that load video to be played within the application

JamesSanga committed:
Library that supports youtube videos to play to mobile applications

JamesSanga committed:
Youtube API key that helps to load videos in application

JamesSanga committed:
Activity that plays trailers from youtube within the application

JamesSanga committed:
Library for sliding layout from bottom to top

JamesSanga committed:
View bounds allows image to change width and height

JamesSanga committed:
Dialog interface to display favorite movie one is clicked

JamesSanga committed:
Added initialize method and implementing customizable dialog

JamesSanga committed:
Code cleaning

JamesSanga committed:
Layout that display in dialog to view full screen of favorite movie

JamesSanga committed:
Removing un necessary codes

JamesSanga committed:
Auto generated

JamesSanga committed:
Library for rx java and image view were added

JamesSanga committed:
Linking maven site to download library

JamesSanga committed:
Implementing long click listener in favorite adapter

JamesSanga committed:
Alert dialog to worn user when wants to delete favorite movie, to vie full image when is clicked and implementation of delete method

JamesSanga committed:
Adding notification when retrofit going in error

JamesSanga committed:
Make card view more readable by clearing some warnings

JamesSanga committed:
Including photo viewer in pop up layout

JamesSanga committed:
Youtube api building in grandle properties

JamesSanga committed:
Hiding api key

JamesSanga committed:
Specified location of youtube player in package

JamesSanga committed:
Moved in Youtube player package

JamesSanga committed:
added new package

JamesSanga committed:
Route generated due to the new package introduced

JamesSanga committed:
Youtube api embed in grandle .properties

JamesSanga committed:
Context tool changed due to new package

JamesSanga committed:
New context tool in activity youtube

twalipo commented on app/src/main/java/com/tz/sanga/moviestore/Adapters/FavoriteAdapter.java:
```diff
@@ -16,26 +15,32 @@
-public class FavoriteAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+public class FavoriteAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>{
     private static final String BASE_URL_IMG = "https://image.tmdb.org/t/p/original";
```
Why not have a constant file where you keep all your constants

twalipo commented on app/src/main/java/com/tz/sanga/moviestore/Adapters/MoviesAdapter.java:
```diff
@@ -64,28 +68,28 @@ public MoviesAdapter(Context context) {
+                                + " | "+ movieVH.movie.getOriginalLanguage().toUpperCase());
+                movieVH.textViewVoteAverage.setText("Vote average" + ": "+String.valueOf(movieVH.movie.getVoteAverage()));
                 //load images by glide library
-                Glide .with(context).load(BASE_URL_IMG + result.getPosterPath())
```
You can still shift this concatenation logic in the specific model  - Create a method that does this in the model

twalipo commented on app/src/main/java/com/tz/sanga/moviestore/Adapters/TrailerAdapter.java:
```diff
@@ -0,0 +1,67 @@
+        return moveData.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
```
You can have the viewHolder implement onClickListener
```suggestion
    public class ViewHolder extends RecyclerView.ViewHolder implement OnClickListener{
        @BindView(R.id.name_trailer) TextView textView;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int position = getAdapterPosition();
            if (position != RecyclerView.NO_POSITION){
                Intent intent = new Intent(context, YoutubePlayer.class);
                intent.putExtra("key", moveData.get(position).getKey());
                intent.putExtra("name", moveData.get(position).getName());
                context.startActivity(intent);
            }
        }
    }
```

JamesSanga committed:
Implemented due several repetition of base url for image

JamesSanga committed:
Included implements on click and on long click listener

JamesSanga committed:
Replaced base url for image with constants that define the url and being used over all place needed

JamesSanga committed:
Declare contex as private

JamesSanga committed:
changes without no new was added

JamesSanga committed:
New method is created for all initialize and notify data changed to adapter

JamesSanga committed:
Making contex private

JamesSanga committed:
Unused imports removed

JamesSanga committed:
Removing some warnings by adding @NonNull in some methods

JamesSanga committed:
Made view holder implements on click listener

JamesSanga committed:
Calling the youtube api

JamesSanga committed:
Erased due to duplicate of codes

JamesSanga committed:
Getting youtube api

JamesSanga committed:
Network state permissions

JamesSanga committed:
Model, view, viewModel fragment

JamesSanga committed:
Changing size of image button

JamesSanga committed:
Adding life cycle components and room components

JamesSanga committed:
Removing warnings

JamesSanga committed:
UI for mvvp fragment

JamesSanga committed:
Adding and implementation for navigation from host fragment to mvvp fragment

JamesSanga committed:
Changed string updated

JamesSanga committed:
Mvvp menu item added in main menu to navigate from host fragment to mvvm fragment (Blank fragment)

JamesSanga committed:
Removing card elevation and make card radius one

JamesSanga committed:
Action and fragment added (blank fragment)

JamesSanga committed:
Note properties in Note class

JamesSanga committed:
Is an interface for some methods

JamesSanga committed:
A class that build instance if is null and populate database

JamesSanga committed:
Class that make changes on the background

JamesSanga committed:
Class that connect updates any changes to ui

JamesSanga committed:
String for MVVM

JamesSanga committed:
auto generated

JamesSanga committed:
Update README.MD

JamesSanga committed:
Local database adapter for data access object (DAO) in room database

JamesSanga committed:
Layout made to display large image after being clicked

JamesSanga committed:
UI for room database implementation in local database

JamesSanga committed:
updating library and removing duplicated library

JamesSanga committed:
UI for DAO

JamesSanga committed:
An interface for room database queries

JamesSanga committed:
Local database and its entities are created here

JamesSanga committed:
Local database adapter for data access objec (DAO) in room database

JamesSanga committed:
Adding Constant class for constants

JamesSanga committed:
Declaring all attributes for table entity

JamesSanga committed:
Repository assists to fetch data from and save to database

JamesSanga committed:
Class that act as an intermediate between model and UI for all changes it updates to both directions

JamesSanga committed:
Checking if path is null or not null by logd method

JamesSanga committed:
Recycler view text view to display if no data exist

JamesSanga committed:
tools list item and colour ascent in text view were added

JamesSanga committed:
Tools item added

JamesSanga committed:
Tool item added

JamesSanga committed:
Live data in host fragment implemented

JamesSanga committed:
Class that updates model and UI

JamesSanga committed:
make access private for arrayList getResults method tha is accessible with other class

JamesSanga committed:
Menu in model view view-model

JamesSanga committed:
Arguments to be moved as bundle defined blank fragment2 layout

JamesSanga committed:
Removed classes after being utilized

JamesSanga committed:
Four strings defined for usability where needed

JamesSanga committed:
Comining changes together
Merge branch 'DaggerInjection' of https://github.com/JamesSanga/MovieStore into DaggerInjection

JamesSanga committed:
Classified into UI

JamesSanga committed:
Classified into UI

JamesSanga committed:
Changes from package UI

JamesSanga committed:
Moved to UI package

JamesSanga committed:
Updated gradle version from 3.4.0 to 3.4.1

JamesSanga committed:
Three constants added (path, FavoriteDatabase and Repository)

JamesSanga committed:
Deleted delete favorite menu

JamesSanga committed:
Removed classes due to migration to room database

JamesSanga committed:
Renamed class from adapter to Favorite adapter

JamesSanga committed:
New query added and other modified

JamesSanga committed:
Removed to be replaced with room db

JamesSanga committed:
making database allowing main threads queries

JamesSanga committed:
Data source class tha implements interface data source created

JamesSanga committed:
Moved to UI Package

JamesSanga committed:
Moved to local into Database package

JamesSanga committed:
Added into repositories package

JamesSanga committed:
Removed check favorite method

JamesSanga committed:
Moved to Presenters package

JamesSanga committed:
Moved to presenters package

JamesSanga committed:
Renamed to favorite fragment

JamesSanga committed:
removed

JamesSanga committed:
Context changed due to addition package UI

JamesSanga committed:
Changed context due to new packages

JamesSanga committed:
Moved to Presenters package

JamesSanga committed:
Moved to Presenters package

JamesSanga committed:
Interface class added

JamesSanga committed:
Removed some item in main menu

JamesSanga committed:
Moved to DI package

JamesSanga committed:
Moved to DI package

JamesSanga committed:
MVVM menu item add note removed

JamesSanga committed:
All fragments moved to UI package a thing lead to change in name of them in navigation graph

JamesSanga committed:
New class named Repository added

JamesSanga committed:
Moved to  dagger injection (DI)

JamesSanga committed:
Edited string empty view

JamesSanga committed:
happened due to youtube player moved UI

JamesSanga committed:
Auto  generated

JamesSanga committed:
Json file for database version

JamesSanga committed:
Favorite adapter

JamesSanga committed:
UI for favorite adapter

JamesSanga committed:
Path to generate database version history

JamesSanga committed:
Removed Repository as a constant field

JamesSanga committed:
Improving code by adding query to check data if exist

JamesSanga committed:
Removed classes

JamesSanga committed:
Removed class

JamesSanga committed:
Used imports have been removed due to its classes have been deleted

JamesSanga committed:
adding id to constructor

JamesSanga committed:
Construct to Favorite database have been changed its parameter from application to context

JamesSanga committed:
passing path to Favorite repository from first fragment

JamesSanga committed:
Removing constant path in constants

JamesSanga committed:
Unused method in favorite adapter removed

JamesSanga committed:
Query to select id by passing path

JamesSanga committed:
Code arrangement

JamesSanga committed:
Commented some method to get id

JamesSanga committed:
Disabled method to get id

JamesSanga committed:
Creating favorite repository

JamesSanga committed:
code arrangement

JamesSanga committed:
Refactored from FavoriteNote to favorite

JamesSanga committed:
FavoriteNote changed due to FavoriteNote class renamed to Favorite

JamesSanga committed:
Renamed favorite note to favorite and changed delete query that passing path as string where to be deleted

JamesSanga committed:
Renamed favorite note to favorite

JamesSanga committed:
Effect of renamed class favorite note to favorite

JamesSanga committed:
Implemented drawable animation loading

JamesSanga committed:
Affected variables due to changed favorite note class to favorite

JamesSanga committed:
images for drawable animation

JamesSanga committed:
Changing image view background

JamesSanga committed:
Renamed from trailerplayer to trailer_player

JamesSanga committed:
Removed images

JamesSanga committed:
Implemented drawable animation

JamesSanga committed:
Changed background of recycler view

JamesSanga committed:
Default change
